client/js/application.ts

```diff
@@ -19,11 +22,16 @@ submitButton.addEventListener("click", e => {
 		body: new FormData(form)
 	}).then(checkStatus).then(parseJSON).then(async () => {
 		let successMessage: string = formType === FormType.Application ?
-			"Your application has been saved. Feel free to come back here and edit it at any time." :
+			"Your application has been saved." + (!unauthenticated ? "Feel free to come back here and edit it at any time." : "") :
 			"Your RSVP has been saved. Feel free to come back here and edit it at any time. We look forward to seeing you!";
 
 		await sweetAlert("Awesome!", successMessage, "success");
-		window.location.assign("/");
+
+		if (unauthenticated) {
+			(document.querySelector("form") as HTMLFormElement).reset();
```

Comment:
Cast not needed here. Typescript can tell that searching for "form" will always return a form element

Reply:
The issue is that the type returned is `HTMLFormElement | null` and I want to force it to be not null.